File: HomeworkDay06/app/src/main/java/com/example/homework_day_06/ItemPhoneImageActivity.java
package com.example.homework_day_06;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

import com.example.homework_day_06.event.BackImageEvent;
import com.example.homework_day_06.event.ImageEvent;

import org.greenrobot.eventbus.EventBus;
import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;

public class ItemPhoneImageActivity extends AppCompatActivity {

    private static final String TAG = "ItemPhoneImageActivity";

    private int position;

    private boolean loveStatus;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_phone_image);
        EventBus.getDefault().register(this);

        Button button = findViewById(R.id.item_love_status);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                loveStatus = !loveStatus;
                EventBus.getDefault().post(new BackImageEvent(loveStatus, position));
                if(loveStatus) {
                    button.setBackgroundColor(getResources().getColor(R.color.coral));
                }
                else {
                    button.setBackgroundColor(getResources().getColor(R.color.gray));
                }
            }
        });
    }

    @Subscribe(threadMode = ThreadMode.BACKGROUND, sticky = true)
    public void onMessageEvent(ImageEvent imageEvent){
        position = imageEvent.getPosition();
        runOnUiThread(() -> {
            ImageView imageView = findViewById(R.id.image);
            imageView.setImageResource(imageEvent.getImageId());
            Button button = findViewById(R.id.item_love_status);
            loveStatus = imageEvent.getLoveStatus();
            if(loveStatus) {
                button.setBackgroundColor(getResources().getColor(R.color.coral));
            }
            else {
                button.setBackgroundColor(getResources().getColor(R.color.gray));
            }
        });
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        EventBus.getDefault().unregister(this);
    }
}

File: HomeworkDay11/app/src/main/java/com/example/homework_day_11/ItemPhoneImageActivity.kt
package com.example.homework_day_11

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.example.homework_day_11.fragment.HomeFragment
import org.greenrobot.eventbus.EventBus

class ItemPhoneImageActivity : AppCompatActivity() {
    private val layoutPosition by lazy {
        intent.getIntExtra("layoutPosition", -1)
    }

    private var isLove = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_item_phone_image)

        val imageResourceId = intent.getIntExtra("imageResourceId", 0)
        val imageView = findViewById<ImageView>(R.id.item_image)
        Glide.with(imageView.context)
            .load(imageResourceId)
            .into(imageView)
        isLove = intent.getBooleanExtra("isLove", false)
        val button = findViewById<TextView>(R.id.toChangeLoveStatus)
        val color = if (isLove) {
            resources.getColor(R.color.coral)
        } else {
            resources.getColor(R.color.gray)
        }
        button.setBackgroundColor(color)
        button.setOnClickListener {
            isLove = !isLove;
            val color1 = if (isLove) {
                resources.getColor(R.color.coral)
            } else {
                resources.getColor(R.color.gray)
            }
            button.setBackgroundColor(color1)
            EventBus.getDefault().post(HomeFragment.LoveStatusEvent(layoutPosition, isLove))
        }
    }
}